cmd/argo/commands/cp.go

```diff
@@ -0,0 +1,143 @@
+				nodeInfo := workflow.Status.Nodes.Find(func(n v1alpha1.NodeStatus) bool { return n.ID == artifact.NodeID })
+				if nodeInfo != nil {
+					customPath = strings.Replace(customPath, "{templateName}", nodeInfo.TemplateName, 1)
+				}
+				customPath = strings.Replace(customPath, "{namespace}", namespace, 1)
+				customPath = strings.Replace(customPath, "{workflowName}", workflowName, 1)
+				customPath = strings.Replace(customPath, "{nodeId}", artifact.NodeID, 1)
+				customPath = strings.Replace(customPath, "{artifactName}", artifact.Name, 1)
+				err = os.MkdirAll(customPath, os.ModePerm)
+				if err != nil {
+					return fmt.Errorf("failed to create folder path: %w", err)
+				}
+				key, err := artifact.GetKey()
+				if err != nil {
+					return fmt.Errorf("error getting key for artifact: %w", err)
+				}
+				err = getAndStoreArtifactData(namespace, workflowName, artifact.NodeID, artifact.Name, path.Base(key), customPath, c, client.ArgoServerOpts)
+				if err != nil {
+					return fmt.Errorf("failed to get and store artifact data: %w", err)
+				}
+			}
+			return nil
+		},
+	}
+	command.Flags().StringVarP(&cpArtifactOpts.namespace, "namespace", "n", "", "namespace of workflow")
+	command.Flags().StringVar(&cpArtifactOpts.nodeId, "node-id", "", "id of node in workflow")
+	command.Flags().StringVar(&cpArtifactOpts.templateName, "template-name", "", "name of template in workflow")
+	command.Flags().StringVar(&cpArtifactOpts.artifactName, "artifact-name", "", "name of output artifact in workflow")
+	command.Flags().StringVar(&cpArtifactOpts.path, "path", "{namespace}/{workflowName}/{nodeId}/outputs/{artifactName}", "use variables {workflowName}, {nodeId}, {templateName}, {artifactName}, and {namespace} to create a customized path to store the artifacts; example: {workflowName}/{templateName}/{artifactName}")
+	return command
+}
+
+func getAndStoreArtifactData(namespace string, workflowName string, nodeId string, artifactName string, fileName string, customPath string, c *http.Client, argoServerOpts apiclient.ArgoServerOpts) error {
+	request, err := http.NewRequest("GET", fmt.Sprintf("%s/artifacts/%s/%s/%s/%s", argoServerOpts.GetURL(), namespace, workflowName, nodeId, artifactName), nil)
+	if err != nil {
+		return fmt.Errorf("failed to create request: %w", err)
+	}
+	request.Header.Set("Authorization", client.GetAuthString())
+	resp, err := c.Do(request)
+	if err != nil {
+		return fmt.Errorf("request failed with: %w", err)
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != 200 {
+		return fmt.Errorf("request failed %s", resp.Status)
+	}
+	artifactFilePath := filepath.Join(customPath, fileName)
+	fileWriter, err := os.Create(artifactFilePath)
+	if err != nil {
+		return fmt.Errorf("creating file failed: %w", err)
+	}
+	defer fileWriter.Close()
+	_, err = io.Copy(fileWriter, resp.Body)
+	if err != nil {
+		return fmt.Errorf("copying file contents failed: %w", err)
+	}
+	log.Printf("Stored artifact %s", fileName)
```

Comment:
Can you change this to `"Created %q`, filename"

Reply:
Updated